kuma/static/js/components/modal.js

```diff
@@ -0,0 +1,88 @@
+(function() {
+    'use strict';
+
+    if (!window.mdn) {
+        window.mdn = {};
+    }
+
+    /**
+     * A modal dialog implemetation based on:
+     * https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+     */
+    var modalDialog = {
+        /**
+         * Returns `false` if either of the following conditions are true else, returns `true`:
+         * 1. Multi auth is not turned on
+         * 2. The modal we expect is not present
+         * 3. We are on a mobile phone
+         * @param {Object} modal - The modal as an `HTMLElement`
+         * @returns {Boolean}
+         */
+        showModal: function (modal) {
+            if (
+                !window.mdn.multiAuthEnabled ||
+                !modal ||
+                (window.matchMedia &&
+                    window.matchMedia('(max-width:47.9385em)').matches)
```

Comment:
So, this means that as soon as this JS code is loaded (which is basically as soon as the JS has been downloaded or cached) it executes a `window.matchMedia()` even though the DOM might not have been painted yet. 
Also, can't you just use `clientWidth` instead? It's bound to be [order of magnitude faster](https://www.measurethat.net/Benchmarks/Show/1213/3/matchmedia-vs-clientwidth#latest_results_block)

Reply:
Also, 99.9% of uses don't click "Sign in". For all of these 99.9% of people, they have to delay their page rendering for this DOM operation to complete first. 

What it ought to be is that this stuff is figured out in runtime. I.e. 
```javascript
document.querySelector('a.signin-linlk').addEventListener('click', event=> {
  if (isModalEnabled()) {
      event.preventDefault()
     showModal();
  }
});
```